### PR TITLE
Add slider handle snap

### PR DESCRIPTION
**Pull request by kaadmy**

### CODE REVIEW

File: Sources/ObjectivelyMVC/Slider.c [hunk 1]
```diff
@@ -161,9 +161,13 @@ static _Bool captureEvent(Control *self, const SDL_Event *event) {
 			if (frame.w) {
 
 				const double fraction = (event->motion.x - frame.x) / (double) frame.w;
-				const double value = this->min + (this->max - this->min) * clamp(fraction, 0.0, 1.0);
+				double value = this->min + (this->max - this->min) * clamp(fraction, 0.0, 1.0);
 
-				$(this, setValue, value); // FIXME: Factor step into this
+				if (this->snapToStep) {
+					value = clamp(round(value / this->step) * this->step, this->min, this->max);
```
Comment (jdolan) [inline Sources/ObjectivelyMVC/Slider.c:167]: This definitely won't work for decimal values / decimal steps.

File: Sources/ObjectivelyMVC/Slider.c [hunk 2]
```diff
@@ -202,6 +206,8 @@ static Slider *initWithFrame(Slider *self, const SDL_Rect *frame, ControlStyle s
 		self->label->view.alignment = ViewAlignmentMiddleRight;
 		self->label->view.padding.left = DEFAULT_SLIDER_LABEL_PADDING;
 
+		self->snapToStep = DEFAULT_SLIDER_SNAP_TO_STEP;
```
Comment (jdolan) [inline Sources/ObjectivelyMVC/Slider.c]: No need to even include this. All instances in Objectively are initialized with `0`, so for things that default to false, I don't even mention them in the initializers.

File: Sources/ObjectivelyMVC/Slider.h [hunk 3]
```diff
@@ -113,6 +114,11 @@ struct Slider {
 	double step;
 
 	/**
+	 * @brief Enable snapping the sniper handle to the step
```
Comment (jdolan) [inline Sources/ObjectivelyMVC/Slider.h]: "slider" not "sniper" LOL Freudian slip?

**Review by jdolan — CHANGES_REQUESTED** [review]

File: Sources/ObjectivelyMVC/Slider.c (hunk 1 above)
Comment (kaadmy) [reply Sources/ObjectivelyMVC/Slider.c:167]: I thought they were always doubles?

**Edit**: yeah, I forgot about that.

**Edit2:** I'm confused...

File: Sources/ObjectivelyMVC/Slider.c (hunk 2 above)
Comment (kaadmy) [reply Sources/ObjectivelyMVC/Slider.c]: I had that set to `true` while testing, it's just easier for me.

File: Sources/ObjectivelyMVC/Slider.h (hunk 3 above)
Comment (kaadmy) [reply Sources/ObjectivelyMVC/Slider.h]: Yeah it was a Fowardian slip.

File: Sources/ObjectivelyMVC/Slider.c (hunk 1 above)
Comment (kaadmy) [reply Sources/ObjectivelyMVC/Slider.c:167]: Unless I'm getting overly confused by this, shouldn't it work?
Did some tests in Python to make sure I'm not going insane:
```
>>> min(max(round(5.6 / 0.5) * 0.5, 1.0), 30.0)
5.5
>>> min(max(round(5.8 / 0.5) * 0.5, 1.0), 30.0)
6.0
>>> min(max(round(8.3 / 0.3) * 0.3, 1.0), 30.0)
8.4
```

As far as I can tell, this should work correctly.

**Comment by kaadmy** [comment]
Ping @jdolan?

**Pull request merged**